docs/src/charm/howto/ceph-csi.md

```diff
@@ -0,0 +1,150 @@
+# ceph-csi
+
+[Ceph] can be used to hold Kubernetes persistent volumes and is the recommended
+storage solution for Canonical Kubernetes.
+
+The ``ceph-csi`` plugin automatically provisions and attaches the Ceph volumes
+to Kubernetes workloads.
+
+Follow this guide to find how Canonical Kubernetes can be integrated with Ceph
+through Juju.
+
+## Prerequisites
+
+This guide assumes that you have an already existing Canonical Kubernetes
+cluster. See the [charm installation] guide for more details.
+
+## Deploying Ceph
+
+We'll deploy a Ceph cluster containing one monitor and three storage units
+(OSDs). For the purpose of this demo, we'll allocate a limited amount of
+resources.
+
+```
+juju deploy -n 1 ceph-mon \
+    --constraints "cores=2 mem=4G root-disk=16G" \
+    --config monitor-count=1
+juju deploy -n 3 ceph-osd \
+    --constraints "cores=2 mem=4G root-disk=16G" \
+    --storage osd-devices=1G,1 --storage osd-journals=1G,1
+juju integrate ceph-osd:mon ceph-mon:osd
+```
+
+If Juju is configured to use the localhost/LXD cloud, please add
+the following constraint to the osd and k8s units:
+``virt-type=virtual-machine``. The k8s unit should then also be configured to
+use a VM container with ubuntu 22.04 as base:
+
+```
+juju deploy k8s --channel=latest/edge \
+    --base "ubuntu@22.04" \
+    --constraints "cores=2 mem=8G root-disk=16G virt-type=virtual-machine"
```

Comment:
The prerequisite states you already have a canonical kubernetes cluster running. I would add it as a prerequisite that if running with localhost/LXD  the flag virt-type=virtual-machine be added to the k8s VM container